src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php

```diff
@@ -46,13 +47,23 @@ public function logKernelException(ExceptionEvent $event)
     {
         $throwable = $event->getThrowable();
         $logLevel = null;
+
         foreach ($this->exceptionsMapping as $class => $config) {
-            if ($throwable instanceof $class && $config['log_level']) {
+            if (!$throwable instanceof $class) {
+                continue;
+            }
+            if ($config['log_level'] && null === $logLevel) {
                 $logLevel = $config['log_level'];
+            }
+            if ($config['status_code'] && $event->getThrowable() === $throwable) {
+                $throwable = new HttpException($config['status_code'], $throwable->getMessage(), $throwable);
```

Comment:
I would also pass the headers argument when provided. I mean, if `$throwable` is originally an instance of `HttpExceptionInterface` and a custom header is set then it should be taken into account during the error rendering response, right?

Reply:
Maybe we should just skip decorating instances of of HttpExceptionInterface? I don't see the point of reconfiguring those.

Reply:
I updated the PR this way. Feedback welcome.

Reply:
I don't think we should exclude any exception, regardless of the headers edge-case, we might want to change the status code of an HTTP exception that we do not control on the app side.

Reply:
In my opinion, the previous code was correct, it was just missing setting the headers when provided.